### DISCUSSION TITLE
Create lambda using aws cdk with 2 different language in one project

### DISCUSSION THREAD

**Discussion by wapt49** [aws/aws-cdk#22982] [General]
Hi,

I need to create lambda using 2 different language in one project cdk.
1. Typescript (for general purpose)
2. Python (for data engineering that use python library like panda, numpy, etc)

I expect if I do something like cdk deploy.. all of my lambda will be deployed using it's environment
Is it possible?

**Comment by indrora** [comment]
You can easily write constructs in Typescript and then (Through JSII) interact with them in Python. 

I am not aware of any polylingual projects, but I'm sure it *could* be done with the right level of ingenuity. 

As for lambda functions: Lambda functions run on the runtime specified in their construct properties. You are not limited to using Javascript in your AWS Lambda functions if you have built your CDK application in Typescript, nor are you required to use Python for your functions if you have developed your CDK app in Python.
